### ISSUE TITLE
Show warning for BSAs with wrong versions

### ISSUE THREAD

**Issue by Infernio** [wrye-bash/wrye-bash#489]
*Basically just putting this here as a personal TODO, it's something I want to add.*

## TL;DR

Seeing as we show a warning for form version 43 plugins on SSE, for localized plugins without strings files, mods or saves with an invalid version, etc., I think we should show one for mismatched BSA versions on boot/refresh as well.

## Details

BSAs have a version field, with each game (except Morrowind) having one specific version that they are compatible with. They generally indicate a major change in the BSA format (e.g. FO3 introduces embedded filenames, SSE uses lz4 and has 64-bit offsets, etc.). A full table follows here for completeness, but this is all already contained in `bsa_files.py`:
| Game | BSA Version |
| --- | --- |
| Morrowind | N/A |
| Oblivion | `0x67` |
| Fallout 3, Fallout NV, Skyrim, Enderal | `0x68` |
| Skyrim SE, Skyrim VR | `0x69` |
| Fallout 4, Fallout 4 VR | `0x01` |

Note: It probably went back down to `0x01` for FO4 because of the switch to an entirely new format (BA2).

If you use a BSAs with a version that is incompatible with the game's version, bad stuff:tm: happens. I've seen crashes upon launching the game, as well as the game getting stuck on the main menu (with the New Game etc. buttons never appearing).

The most common way for this to happen is if someone forgets to unpack and repack a Skyrim LE BSA to the Skyrim SE format when porting, but this is theoretically going to be a problem for all games, plus a general warning is going to be easier; so I propose to do this for all games.

**Comment by Infernio** [comment]
![BSAWarning](https://user-images.githubusercontent.com/2845240/76703664-58abad00-66d3-11ea-9673-7f52652246aa.png)

**Comment by Infernio** [comment]
Oh wow, testing the commit lead me down some rabbit hole:
 1. Our ArchiveInvalidationInvalidated BSAs for FO3 and FNV have the wrong version, they're actually Oblivion BSAs
 2. Our BSA Redirection implementation tries to undo modifications to `Oblivion - Textures - Compressed.bsa`, even for FO3/FNV, where it obviously doesn't exist - completely broken and crashes
I asked on the xEdit server if FO3/FNV BSAs can load Oblivion BSAs as well - if not, we've been putting broken BSAs into people's installations for years 🙃

**Comment by Utumno** [comment]
Yep me and you falling through rabbit holes (my latest was status bar drag and drop, had bugs since forever) shows basically that we should really start wrapping up at this point - test and finalize the wx3/4 and prepare for python 3 (the unicode strings is the next big thing and we really want to see if we want to add 25k u prefixes). We have really shaken the foundations of the code with wx rewrite - surely bugs would be revealed, we should now sit back and wrap up a bit :)

**Comment by Arthmoor** [comment]
Don't you guys wish you'd taken the blue pill now? :P

**Comment by Utumno** [comment]
Color blindness :P

**Comment by Infernio** [comment]
Note to self: maybe whitelist Oblivion BSAs for Skyrim LE as well? People on Discord are reporting that the game is working fine for them with it...

**Comment by Arthmoor** [comment]
I'd recommend against that as there are probably enough differences in how Skyrim LE handles BSA files that it may not tolerate being marked as an Oblivion BSA in all cases.

Also was at least one person I saw on Discord saying that it caused his game to crash with a bad version on one.
